Added first graphic

jhdalek55 opened the pull request:
I  have just begun to add the figures into the file. This is a work in progress, so do not review or merge this pull request just yet.

jhdalek55 commented:
@tkfu and @patrickvacek. I submitted this PR to make sure I was placing the graphics correctly. Can one of you check this for me and just tell me if what I have done here works? I didn't want to place the graphics and then have to re-do them if they were wrong. So if this work I can go ahead and get the others loaded.

patrickvacek commented on customizations.md:
```diff
@@ -30,7 +33,7 @@ Finally, in order to install a delta image, an ECU SHOULD take one of the action
 ### TODO Recreate and insert Table D.2a from google doc and label it as Table 1.
 
-
+                                                                                                                                                                                                                 |
```
This weird little extra change seems superfluous.

jhdalek55 replied:
Fixed this

patrickvacek commented on customizations.md:
```diff
@@ -25,113 +23,124 @@ Examples of data include setup or initialization data, such as engine parameters
-In the first option, an OEM MAY elect to receive input from a third party as to which updates should be installed. The process is illustrated in Figure 3. In the first step, the vehicle would submit to the director repository controlled by the OEM its vehicle version manifest, which lists which updates are currently installed. In the second step, the director repository would perform dependency r ...
+In the first option, an OEM MAY elect to receive input from a third party as to which updates should be installed. The process is illustrated in Figure 3. In the first step, the vehicle would submit its vehicle version manifest to the Director repository controlled by the OEM. The manifest lists which updates are currently installed. In the second step, the Director repository would perform depen ...
+
+### TODO Write rationale for the SHOULD noted above.
```
I think it's covered with the "so that ..." bit.

jhdalek55 commented on customizations.md:
```diff
@@ -25,113 +23,124 @@ Examples of data include setup or initialization data, such as engine parameters
 ### Dynamic delta updates vs. precomputed delta updates
 
-There are two options when computing delta updates. Delta updates can be computed dynamically for each ECU during the installation process (dynamic delta updates), or possible delta images can be precomputed before installation begins (precomputed delta updates). The process for describing both types of updates appears below in the subsection on custom installation instructions.
+There are two options when computing delta updates. Delta updates can be computed dynamically for each ECU during the installation process (dynamic delta updates), or possible delta images can be precomputed before installation begins (precomputed delta updates). The process for describing both types of updates appears below in the subsection on Custom installation instructions.
```
I capitalized it because its the name of a subsection on another page to which this will be links. Subheads. I'm trying to be consistent with the style used in the Standard in which subsection titles capitalized just the first word. I'm still not quite sure how to do links to other sections. I'm assuming the text will be highlighted in some manner when this is done and then it will not look so weird.

jhdalek55 replied:
If its just inserting a simple link, i think I can take care of that. I'm not sure about linking to a particular subsection. I think I would rather get everything else committed and then do the links by themselves.

patrickvacek replied:
As you like. But if you approve my suggested change here, then the work in this case is already done. :)

jhdalek55 commented on customizations.md:
```diff
@@ -25,113 +23,124 @@ Examples of data include setup or initialization data, such as engine parameters
+If multiple Primaries are active in the vehicle at the same time, then each Primary SHOULD control a mutually exclusive set of Secondaries, so that each Secondary is controlled by one Primary.
 
-If multiple primaries are active in the vehicle at the same time, then each primary SHOULD control a mutually exclusive set of secondaries, so that each secondary is controlled by one primary.
+### TODO Write rationale for the SHOULD noted above.
```
These TODO messages come at the request of Ira who suggested that eventually every "should" in the deployment pages requires a "rationale" statement  as to why we are making this recommendation.

patrickvacek replied:
I understand the motivation, but what more rationale is required than what's already there?

jhdalek55 commented:
@trishankkarthik can you give this a quick look. I'd like to close this out.

trishankatdatadog commented:
@jhdalek55: This PR is more than just adding a graphic. Could you please roughly describe what it's about, and what I should look for? Thanks!

trishankatdatadog commented:
@jhdalek55 Could you please replace all `“` with `"`?

jhdalek55 commented:
Trishank..SO I didn't have too many pull requests I added on to the "adding graphic PR." I am doing the following--1) adding graphics, 2) working on capitalization consistency 3) replacing MUSTs will SHOULDs and 4) Where editing is desperately cried out for, doing some sentence level editing. This is all on the "customization" page.

jhdalek55 commented:
Nicely nudging all involved. I believe I have addressed all the necessary comments. Can I merge?

jhdalek55 commented:
I changed the header--which I should have done when I added to it. This is still PR#9 that has been reviewed by two people. If its ok with @tkfu I plan to merge.

patrickvacek commented on customizations.md:
```diff
@@ -25,113 +23,124 @@ Examples of data include setup or initialization data, such as engine parameters
+An OEM may wish to require atomic installation of a bundle of images, which means that either all updates in a bundle are installed, or, should one or more fail, none of them are installed. Uptane does not provide a way to guarantee such atomic installation because the problem is out of its scope. It is challenging for ECUs to atomically install a bundle in the face of arbitrary failure: if just  ...
 
-The simplest solution is to use the vehicle version manifest to report to the director repository the failure to atomically install a bundle, and not to retry installation. After receiving the report, it is up to the OEM to decide how to respond. For example, the OEM MAY require the owner of the vehicle to diagnose the failure at the nearest dealership or authorized mechanic.
+The simplest solution is to use the vehicle version manifest to report any failure to atomically install a bundle to the Director repository, and then not retry installation. After receiving the report, it is up to the OEM to decide how to respond. For example, the OEM MAY require the owner of the vehicle to diagnose the failure at the nearest dealership or authorized mechanic.
```
```suggestion
The simplest solution is to use the vehicle version manifest to report to the Director repository any failure to atomically install a bundle, and then not retry installation. After receiving the report, it is up to the OEM to decide how to respond. For example, the OEM MAY require the owner of the vehicle to diagnose the failure at the nearest dealership or authorized mechanic.
```

jhdalek55 commented:
Patrick, I'll see if I can get a clarification from Ira today as to whether these "shoulds" are to be a separate footnote or a "call out" box, or if the explanations in the text are sufficient. I would not particularly like to keep these TODOs in any longer than I have to.

Jon is still listed as a "suggested reviewer"so if he has the time I'd like him to take a look. I think otherwise this is good to go.

tkfu commented:
I don't quite have the time to do a detailed review here, but it seems like @patrickvacek and @trishankatdatadog have gone over it pretty thoroughly.

jhdalek55 commented:
I am going to merge this then. The one outstanding issue Patrick identified will be addressed when we get a clarification on what "explanations of shoulds" must consider.
